addons/web/static/src/legacy/js/fields/basic_fields.js

```diff
@@ -607,6 +607,13 @@ var FieldChar = InputField.extend(TranslatableFieldMixin, {
         }
         return this._super(value, options);
     },
+
+    isValid: function () {
+        if (this.value && this.field.pattern && !(new RegExp(this.field.pattern)).test(this.value)) {
```

Comment:
Python and JS regex overlap but they are far from identical. A regexp which works on the Python side (and is actively useful because the value is set through a process other than a form view) can throw an exception here.

Reply:
We already discussed this (you included in the conversation iirc) and agreed that the necessary subset was enough.

Reply:
> We already discussed this (you included in the conversation iirc) and agreed that the necessary subset was enough.

The problem is not that "the common subset is fine", it's that this code is broken. A developer can define a pattern on a field which appears in no form view, then somebody else decides the form view is useful after all and now the client raises an exception every time it tries to validate the field.